src/libsyntax/attr/mod.rs

```diff
@@ -41,51 +41,31 @@ use std::iter;
 
 pub fn mark_used(attr: &Attribute) {
     debug!("Marking {:?} as used.", attr);
-    let AttrId(id) = attr.id;
     GLOBALS.with(|globals| {
         let mut slot = globals.used_attrs.lock();
-        let idx = (id / 64) as usize;
-        let shift = id % 64;
-        if slot.len() <= idx {
-            slot.resize(idx + 1, 0);
-        }
-        slot[idx] |= 1 << shift;
+        slot.insert(attr.id);
```

Comment:
Would you mind changing this (and the other cases below) to just `globals.used_attrs.lock().insert(attr.id);`? The `slot` binding does not really add much to the codes readability.